src/PythonClient/src/quixstreams/models/streamproducer/streameventsproducer.py

```diff
@@ -35,58 +34,62 @@ def __init__(self, net_pointer: ctypes.c_void_p):
 
     def flush(self):
         """
-        Immediately writes the event definitions from the buffer without waiting for buffer condition to fulfill (200ms timeout) #TODO verify 200ms
+        Immediately publishes the event definitions from the buffer without waiting for buffer condition to fulfill
+        (200ms timeout). NOTE: Verify 200ms timeout value.
         """
         self._interop.Flush()
 
     @property
     def default_tags(self) -> Dict[str, str]:
-        """Gets default tags injected to all Events Values sent by the writer."""
+        """Default Tags injected to all Event Values sent by the producer."""
         dic_hptr = self._interop.get_DefaultTags()
         return InteropUtils.invoke_and_free(dic_hptr, lambda x: di.ReadStringStrings(di.ReadAnyHPtrToUPtr(x)), default={})
 
     @property
     def default_location(self) -> str:
         """
-            Gets the default Location of the events. Event definitions added with add_definition  will be inserted at this location.
-            See add_location for adding definitions at a different location without changing default.
-            Example: "/Group1/SubGroup2"
+        Gets the default Location of the events. Event definitions added with add_definition will be inserted at this location.
+        See add_location for adding definitions at a different location without changing default.
+        Example: "/Group1/SubGroup2"
         """
         return self._interop.get_DefaultLocation()
 
     @default_location.setter
     def default_location(self, value: str):
         """
-            Sets the default Location of the events. Event definitions added with add_definition will be inserted at this location.
-            See add_location for adding definitions at a different location without changing default.
-            Example: "/Group1/SubGroup2"
+        Sets the default Location of the events. Event definitions added with add_definition will be inserted at this location.
+        See add_location for adding definitions at a different location without changing default.
+
+        Args:
+            value (str): Location string, e.g., "/Group1/SubGroup2".
         """
         self._interop.set_DefaultLocation(value)
 
     @property
     def epoch(self) -> datetime:
-        """Gets the default epoch used for event values"""
+        """Gets the default epoch used for Timestamp event values."""
 
         ptr = self._interop.get_Epoch()
         value = dtc.datetime_to_python(ptr)
         return value
 
     @epoch.setter
     def epoch(self, value: datetime):
-        """Sets the default epoch used for event values"""
+        """Sets the default epoch used for Timestamp event values. Datetime added on top of all the Timestamps."""
```

Comment:
I don't understand.

"Datetime added on top of all the Timestamps."

Epoch is only used when the event (or timestamps) is added as an offset from epoch (timedelta, nanoseconds), not when added at an exact DateTime.

Example:
epoch 10, ns 300 -> 310 (Added)
epoch 10, 2021/01/06 15:59:36.3213 -> 2021/01/06 15:59:36.3213 (not added)

Reply:
This phrasing doesn't look too bad, maybe can use as base
https://github.com/quixio/quix-streams/blob/6b746a0b7dc60119d87dd2a28ff21f3c246e3b97/src/PythonClient/src/quixstreams/models/timeseriesdataraw.py#L323